File: dart/lib/model/miner_claim_miner_body.dart
part of swagger.api;

class MinerClaimMinerBody {
  
  String claim = null;

  AddressAddress miner = null;

  String name = null;

  MinerClaimMinerBody();

  @override
  String toString() {
    return 'MinerClaimMinerBody[claim=$claim, miner=$miner, name=$name, ]';
  }

  MinerClaimMinerBody.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    claim = json['claim'];
    miner = new AddressAddress.fromJson(json['miner']);
    name = json['name'];
  }

  Map<String, dynamic> toJson() {
    return {
      'claim': claim,
      'miner': miner,
      'name': name
     };
  }

  static List<MinerClaimMinerBody> listFromJson(List<dynamic> json) {
    return json == null ? new List<MinerClaimMinerBody>() : json.map((value) => new MinerClaimMinerBody.fromJson(value)).toList();
  }

  static Map<String, MinerClaimMinerBody> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, MinerClaimMinerBody>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new MinerClaimMinerBody.fromJson(value));
    }
    return map;
  }
}

File: csharp-dotnet2/src/main/CsharpDotNet2/estuary-client/Model/MinerClaimMinerBody.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class MinerClaimMinerBody {
    /// <summary>
    /// Gets or Sets Claim
    /// </summary>
    [DataMember(Name="claim", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "claim")]
    public string Claim { get; set; }

    /// <summary>
    /// Gets or Sets Miner
    /// </summary>
    [DataMember(Name="miner", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "miner")]
    public AddressAddress Miner { get; set; }

    /// <summary>
    /// Gets or Sets Name
    /// </summary>
    [DataMember(Name="name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "name")]
    public string Name { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class MinerClaimMinerBody {\n");
      sb.Append("  Claim: ").Append(Claim).Append("\n");
      sb.Append("  Miner: ").Append(Miner).Append("\n");
      sb.Append("  Name: ").Append(Name).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
